hardware/detect.py

```diff
@@ -51,8 +51,9 @@
 from hardware import rtc
 from hardware import smart_utils
 
-
 SIOCGIFNETMASK = 0x891b
+AUXV_FLAGS = ("AT_HWCAP", "AT_HWCAP2", "AT_PAGESZ",
+              "AT_FLAGS", "AT_PLATFORM")
```

Comment:
I'd like to see AT_BASE_PLATFORM exposed/captured here.  It's meaningful on powerpc:
```
# LD_SHOW_AUXV=1 /bin/true
AT_DCACHEBSIZE:  0x80
AT_ICACHEBSIZE:  0x80
AT_UCACHEBSIZE:  0x0
AT_SYSINFO_EHDR: 0x3fff95170000
AT_HWCAP:        true_le archpmu vsx arch_2_06 dfp ic_snoop smt mmu fpu altivec ppc64 ppc32
AT_PAGESZ:       65536
AT_CLKTCK:       100
AT_PHDR:         0x10000040
AT_PHENT:        56
AT_PHNUM:        9
AT_BASE:         0x3fff95190000
AT_FLAGS:        0x0
AT_ENTRY:        0x1000147c
AT_UID:          0
AT_EUID:         0
AT_GID:          0
AT_EGID:         0
AT_SECURE:       0
AT_RANDOM:       0x3ffff55c9492
AT_HWCAP2:       htm-nosc vcrypto tar isel ebb dscr htm arch_2_07
AT_EXECFN:       /bin/true
AT_PLATFORM:     power8
AT_BASE_PLATFORM:power8
# 
```

However adding that here will mean a tweak to the test as we'd no longer get all the keys in `detect.AUXV_FLAGS` in the hw_list

Reply:
I've done that by introducing AUXV_OPT_FLAGS which may or not be present on a particular architecture.